test/browser/getDomSibling.test.js

```diff
@@ -0,0 +1,353 @@
+import { h, render, Fragment } from '../../src/index';
+import { getDomSibling } from '../../src/component';
+import { setupScratch, teardown } from '../_util/helpers';
+
+/** @jsx h */
+
+describe('getDomSibling', () => {
+
+	/** @type {import('../../src/internal').PreactElement} */
+	let scratch;
+
+	const getRoot = dom => dom._children;
+
+	beforeEach(() => {
+		scratch = setupScratch();
+	});
+
+	afterEach(() => {
+		teardown(scratch);
+	});
+
+	it('should find direct sibling', () => {
+		render((
+			<div>
+				<div>A</div>
+				<div>B</div>
+			</div>
+		), scratch);
+		let vnode = getRoot(scratch)._children[0]._children[0];
+		expect(getDomSibling(vnode)).to.equal(scratch.firstChild.childNodes[1]);
+	});
+
+	it('should find direct text node sibling', () => {
+		render((
+			<div>
+				<div>A</div>
+				B
+			</div>
+		), scratch);
+		let vnode = getRoot(scratch)._children[0]._children[0];
+		expect(getDomSibling(vnode)).to.equal(scratch.firstChild.childNodes[1]);
+	});
+
+	it('should find nested text node sibling', () => {
+		render((
+			<div>
+				<Fragment>
+					<div>A</div>
+				</Fragment>
+				<Fragment>
+					B
+				</Fragment>
+			</div>
+		), scratch);
+		let vnode = getRoot(scratch)._children[0]._children[0];
+		expect(getDomSibling(vnode)).to.equal(scratch.firstChild.childNodes[1]);
+	});
+
+	it('should find text node sibling with placeholder', () => {
+		render((
+			<div>
+				A
+				{null}
+				B
+			</div>
+		), scratch);
+		let vnode = getRoot(scratch)._children[0]._children[0];
+		expect(getDomSibling(vnode)).to.equal(scratch.firstChild.childNodes[1]);
+	});
+
+	it('should find sibling with placeholder', () => {
+		render((
+			<div key="parent">
+				<div key="A">A</div>
+				{null}
+				<div key="B">B</div>
+			</div>
+		), scratch);
+		let vnode = getRoot(scratch)._children[0]._children[0];
+		expect(getDomSibling(vnode)).to.equal(scratch.firstChild.childNodes[1]);
+	});
+
+	it('should find sibling with nested placeholder', () => {
+		render((
+			<div key="0">
+				<Fragment key="0.0">
```

Comment:
I added keys to some of the tests here just cuz they made it easier to debug `getDomSibling` with a couple of well-placed `console.log(vnode.key)`

Reply:
Ohh that's a very neat way to debug ordering. I'll use that on my next PR :+1: